src/ngraph_encapsulate_op.cc

```diff
@@ -168,11 +169,10 @@ class NGraphEncapsulateOp : public OpKernel {
       // TODO(amprocte): We should be able to unref the tracker here, but it
       // seems to screw things up in the C++ unit tests.
       // m_freshness_tracker->Unref();
-
-      // Release the backend
-      BackendManager::ReleaseBackend(m_op_backend_name);
-      NGRAPH_VLOG(2) << "~NGraphEncapsulateOp()";
     }
+    // Release the backend
+    NGRAPH_VLOG(2) << "~NGraphEncapsulateOp():: ReleaseBackend";
+    BackendManager::ReleaseBackend(m_op_backend_name);
```

Comment:
We were releasing the backend only when freshness tracker was not null. It should be released in all the scenarios, so moved it outside the loop.